drivers/gles3/storage/render_scene_buffers_gles3.cpp

```diff
@@ -202,6 +202,10 @@ void RenderSceneBuffersGLES3::_check_render_buffers() {
 
 	if ((!use_internal_buffer || internal3d.color != 0) && (msaa3d.mode == RS::VIEWPORT_MSAA_DISABLED || msaa3d.color != 0)) {
 		// already setup!
+		if (!use_internal_buffer && msaa3d.mode == RS::VIEWPORT_MSAA_DISABLED && (internal3d.color != 0 || msaa3d.color != 0)) {
```

Comment:
I don't think you need to check msaa3d.mode here. MSAA follows its own branch after the internal texture is allocated. The parent if check here checks both MSAA and internal buffer only because it can only early return if neither of them need to be allocated.